### PR TITLE
gh-118406: Add signature for sqlite3.Connection objects

### LINKED ISSUE #118406: Implement inspect.signature() for sqlite3 connection object
# Feature or enhancement

`sqlite3.connection()` returns a custom callable object.  But `inspect.signature()` returns a generic `<Signature (*args, **kwargs)>` for it:
```pycon
>>> import inspect, sqlite3
>>> con = sqlite3.connect(':memory:')
>>> con('select 1')
<sqlite3.Statement object at 0x7f76842d2fd0>
>>> inspect.signature(con)
<Signature (*args, **kwargs)>
```
This issue is similar to #118285 and #118402.

### Linked PRs
* gh-118428

### PR DESCRIPTION
**Pull request by erlend-aasland**
* Issue: gh-118406

### CODE REVIEW

File: Modules/_sqlite/connection.c [hunk 1]
```diff
@@ -2561,6 +2561,12 @@ set_autocommit(pysqlite_Connection *self, PyObject *val, void *Py_UNUSED(ctx))
     return 0;
 }
 
+static PyObject *
+get_sig(pysqlite_Connection *self, void *Py_UNUSED(ctx))
```
Comment (serhiy-storchaka) [inline Modules/_sqlite/connection.c]:
```suggestion
get_sig(PyObject *self, void *Py_UNUSED(ctx))
```
And you will not need the cast.

File: Modules/_sqlite/connection.c [hunk 2]
```diff
@@ -2570,6 +2576,7 @@ static PyGetSetDef connection_getset[] = {
     {"total_changes",  (getter)pysqlite_connection_get_total_changes, (setter)0},
     {"in_transaction",  (getter)pysqlite_connection_get_in_transaction, (setter)0},
     {"autocommit",  (getter)get_autocommit, (setter)set_autocommit},
+    {"__text_signature__", (getter)get_sig, (setter)0},
```
Comment (serhiy-storchaka) [inline Modules/_sqlite/connection.c]:
```suggestion
    {"__text_signature__", get_sig, (setter)0},
```

**Review by serhiy-storchaka — APPROVED** [review]
LGTM. Proposed a minor style change.

**Comment by erlend-aasland** [comment]
Thanks for the review, Serhiy!

**Pull request merged**